### PR TITLE
Adding Simplified to the submission

### PR DESCRIPTION
**Pull request by garethgooch**
updated services to allow for simplified notification to be passed and submitted

### CODE REVIEW

File: test/services/conversion/SimplifiedNotificationConversionServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+
+  private val enRouteEventVehicularTranshipment: Gen[(EnRouteEvent, VehicularTranshipment)] = for {
+    enRouteEvent <- arbitrary[EnRouteEvent]
+    vehicularTranshipment     <- arbitrary[VehicularTranshipment]
+  } yield (enRouteEvent.copy(eventDetails = Some(vehicularTranshipment)), vehicularTranshipment)
+
+
+  "ArrivalNotificationConversionService" - {
+    "return 'Normal Arrival Notification' message" - {
```
Comment (chris-ianson) [inline test/services/conversion/SimplifiedNotificationConversionServiceSpec.scala:70]: Should this not be simplified

File: app/services/conversion/ArrivalNotificationConversionService.scala [hunk 2]
```diff
@@ -30,32 +31,62 @@ class ArrivalNotificationConversionService {
...
+              name            = traderName,
+              streetAndNumber = tradersAddress.buildingAndStreet,
+              postCode        = tradersAddress.postcode,
+              city            = tradersAddress.city,
+              countryCode     = countryCode_GB
+            ),
+            presentationOfficeId   = presentationOffice.id,
+            presentationOfficeName = presentationOffice.name,
+            enRouteEvents          = enRouteEvents(userAnswers)
+          )
```
Comment (chris-ianson) [inline app/services/conversion/ArrivalNotificationConversionService.scala:61]: Could we split out the right hand side of both cases into method calls so the convertToArrivalNotification is easier to read? Or you could use PartialFunctions to separate the case matches (see RemarksServiceImpl in unloading-frontend)

File: app/services/conversion/ArrivalNotificationConversionService.scala [hunk 3]
```diff
@@ -30,32 +31,62 @@ class ArrivalNotificationConversionService {
...
+      case Some(AuthorisedConsigneesLocation) =>
+        for {
+          presentationOffice <- userAnswers.get(PresentationOfficePage)
+          authorisedLocation <- userAnswers.get(AuthorisedLocationPage)
+          tradersAddress     <- userAnswers.get(ConsigneeAddressPage)
+          traderEori         <- userAnswers.get(ConsigneeEoriNumberPage) orElse Some(eoriNumber) //TODO remove .get with answering Eori in journey
+          traderName         <- userAnswers.get(ConsigneeNamePage)
+          notificationPlace  <- userAnswers.get(ConsigneeAddressPage) // orElse Some(tradersAddress)
```
Comment (chris-ianson) [inline app/services/conversion/ArrivalNotificationConversionService.scala:69]: Commented code

File: app/services/conversion/ArrivalNotificationConversionService.scala [hunk 4]
```diff
@@ -30,32 +31,62 @@ class ArrivalNotificationConversionService {
...
+            presentationOfficeId   = presentationOffice.id,
+            presentationOfficeName = presentationOffice.name,
+            enRouteEvents          = enRouteEvents(userAnswers)
+          )
+      case Some(AuthorisedConsigneesLocation) =>
+        for {
+          presentationOffice <- userAnswers.get(PresentationOfficePage)
+          authorisedLocation <- userAnswers.get(AuthorisedLocationPage)
+          tradersAddress     <- userAnswers.get(ConsigneeAddressPage)
+          traderEori         <- userAnswers.get(ConsigneeEoriNumberPage) orElse Some(eoriNumber) //TODO remove .get with answering Eori in journey
```
Comment (chris-ianson) [inline app/services/conversion/ArrivalNotificationConversionService.scala:67]: Commented code

File: test/services/conversion/SimplifiedNotificationConversionServiceSpec.scala (hunk 1 above)
Comment (kenneth-rayner) [reply test/services/conversion/SimplifiedNotificationConversionServiceSpec.scala:70]: Done

File: app/services/conversion/ArrivalNotificationConversionService.scala (hunk 2 above)
Comment (kenneth-rayner) [reply app/services/conversion/ArrivalNotificationConversionService.scala:61]: Spit into separate defs

File: app/services/conversion/ArrivalNotificationConversionService.scala (hunk 3 above)
Comment (kenneth-rayner) [reply app/services/conversion/ArrivalNotificationConversionService.scala:69]: Done

File: app/services/conversion/ArrivalNotificationConversionService.scala (hunk 4 above)
Comment (kenneth-rayner) [reply app/services/conversion/ArrivalNotificationConversionService.scala:67]: Gareth will do a refactor/might be on a separate ticket

Comment (garethgooch) [reply app/services/conversion/ArrivalNotificationConversionService.scala]: Added in saving of eori on the consignee confirmation page

File: app/controllers/ConsigneeEoriConfirmationController.scala [hunk 5]
```diff
@@ -92,7 +92,12 @@ class ConsigneeEoriConfirmationController @Inject()(
               },
               value =>
                 for {
-                  updatedAnswers <- Future.fromTry(request.userAnswers.set(ConsigneeEoriConfirmationPage, value))
+                  ua <- if (value) {
+                    Future.fromTry(request.userAnswers.set(ConsigneeEoriNumberPage, request.eoriNumber))
```
Comment (chris-ianson) [inline app/controllers/ConsigneeEoriConfirmationController.scala:96]: So is this if a user confirms the eori number we set it to ConsigneeEoriNumberPage answer? Do we not need to check the value (whether they've set true or false)

File: app/services/conversion/ArrivalNotificationConversionService.scala [hunk 6]
```diff
@@ -31,64 +31,70 @@ class ArrivalNotificationConversionService {
...
+          postCode        = tradersAddress.postcode,
+          city            = tradersAddress.city,
+          countryCode     = countryCode_GB
+        ),
+        presentationOfficeId   = presentationOffice.id,
+        presentationOfficeName = presentationOffice.name,
+        enRouteEvents          = enRouteEvents(userAnswers)
+      )
+
+  private def createNormalNotification(userAnswers: UserAnswers) =
```
Comment (chris-ianson) [inline app/services/conversion/ArrivalNotificationConversionService.scala:71]: Could we add the return type please (consistency)

File: app/services/conversion/SubmissionModelService.scala [hunk 7]
```diff
@@ -44,7 +44,20 @@ class SubmissionModelService @Inject()() {
...
         ArrivalMovementRequest(meta, header, traderDestination, customsOffice, enRouteEvents)
-      case _ => ???
+
+      case simplifiedNotification: SimplifiedNotification =>
+        val meta = Meta(
+          messageSender               = messageSender,
+          interchangeControlReference = interchangeControlReference,
+          dateOfPreparation           = simplifiedNotification.notificationDate,
+          timeOfPreparation           = timeOfPresentation
+        )
```
Comment (chris-ianson) [inline app/services/conversion/SubmissionModelService.scala:54]: Consider splitting out Meta into a method similar to buildHeader method? (it's duplicate code used in both case statements

File: app/services/conversion/SubmissionModelService.scala [hunk 8]
```diff
@@ -57,6 +70,17 @@ class SubmissionModelService @Inject()() {
       procedureTypeFlag        = procedureTypeFlag,
       notificationDate         = arrivalNotification.notificationDate
     )
+
+  private def buildSimplifiedHeader(arrivalNotification: SimplifiedNotification, procedureTypeFlag: ProcedureTypeFlag): Header =
```
Comment (chris-ianson) [inline app/services/conversion/SubmissionModelService.scala:74]: You could pass in customsSubPlace as an additional method argument and reduce it to single method for building header. I'll let you decide what you want to do with this.

File: app/controllers/ConsigneeEoriConfirmationController.scala (hunk 5 above)
Comment (garethgooch) [reply app/controllers/ConsigneeEoriConfirmationController.scala:96]: The confirm its correct on the page if it is set this on the consigneeEoriNumberPage, if is not correct they are redirected to enter the value

**Pull request closed without merging**